### DISCUSSION TITLE
性能好差，和论文里的不一致

### DISCUSSION THREAD

**Discussion by taoweitao-125** [magicpdf/Magic-PDF#3685] [Q&A] [answered]
### 🔎 Search before asking | 提交之前请先搜索

- [x] I have searched the MinerU [Readme](https://github.com/opendatalab/MinerU) and found no similar bug report.
- [x] I have searched the MinerU [Issues](https://github.com/opendatalab/MinerU/issues) and found no similar bug report.
- [x] I have searched the MinerU [Discussions](https://github.com/opendatalab/MinerU/discussions) and found no similar bug report.

### 🤖 Consult the online AI assistant for assistance | 在线 AI 助手咨询

- [x] I have consulted the [online AI assistant](https://deepwiki.com/opendatalab/MinerU) but was unable to obtain a solution to the issue.

### Description of the bug | 错误描述

我在4090d上和h200上运行，为何延时那么高，远达不到论文里提到的在h200上的4.47pages/s

### How to reproduce the bug | 如何复现

我分别使用了单页的pdf和多页（8页及8页以下）的pdf，解析延时基本在5s-10s之间

<img width="960" height="86" alt="Image" src="https://github.com/user-attachments/assets/1443cf82-c198-4fa2-af9b-cbbf50940526" />

### Operating System Mode | 操作系统类型

Linux

### Operating System Version| 操作系统版本

Ubuntu  2025/09/19 2.5.2 Released

### Python version | Python 版本

3.12

### Device mode | 设备模式

cuda

**Comment by dosubot** [comment]
Hi @taoweitao-125! I'm [Dosu](https://go.dosu.dev/dosubot) and I’m helping the MinerU team.


你遇到的GPU性能问题可能和配置、后端选择、模型版本以及最近的代码变动有关。根据最新的[代码更新](https://github.com/opendatalab/MinerU/pull/3545)，MinerU 2.5.3在VLM后端引入了“custom logits processors”，它们的启用依赖于CUDA可用性、GPU算力（Compute Capability）、vLLM版本，以及环境变量`VLLM_USE_V1`。如果这些条件不满足，可能会导致推理速度远低于论文数据。

此外，环境变量`MINERU_DEVICE_MODE`只对“pipeline”后端有效，VLM后端的设备选择和优化主要受上述新逻辑影响。如果你用的是VLM后端，建议检查：

1. CUDA是否被正确检测到（`torch.cuda.is_available()`为True）。
2. GPU算力是否≥8.0，或vLLM版本≥0.10.2。
3. 环境变量`VLLM_USE_V1`没有被设置为0。
4. 是否使用了最新的MinerU和vLLM版本。

建议你尝试升级到MinerU 2.5.3及以上，并确保环境变量和依赖项配置正确。如果仍然性能低下，可以开启debug日志，确认每一步是否都在GPU上执行，并检查是否有模型未能正确加载到GPU。详细配置和环境变量说明见[相关文档和PR](https://github.com/opendatalab/MinerU/pull/3506)。

如果还有疑问，可以补充你的mineru --version输出、后端类型（pipeline还是vlm）、vLLM版本号，以及相关环境变量设置，这样更容易定位问题。



*To reply, just mention [@dosu](https://go.dosu.dev/dosubot).*

---
<sup>How did I do? [Good](https://app.dosu.dev/response-feedback/ccc2fc1d-c444-4b98-81fe-7a14ee94df02?feedback_type=great_response) | [Irrelevant](https://app.dosu.dev/response-feedback/ccc2fc1d-c444-4b98-81fe-7a14ee94df02?feedback_type=irrelevant_answer) | [Incorrect](https://app.dosu.dev/response-feedback/ccc2fc1d-c444-4b98-81fe-7a14ee94df02?feedback_type=incorrect_sources) | [Verbose](https://app.dosu.dev/response-feedback/ccc2fc1d-c444-4b98-81fe-7a14ee94df02?feedback_type=too_verbose) | [Hallucination](https://app.dosu.dev/response-feedback/ccc2fc1d-c444-4b98-81fe-7a14ee94df02?feedback_type=hallucination) | [Report 🐛](https://app.dosu.dev/response-feedback/ccc2fc1d-c444-4b98-81fe-7a14ee94df02?feedback_type=bug_report) | [Other](https://app.dosu.dev/response-feedback/ccc2fc1d-c444-4b98-81fe-7a14ee94df02?feedback_type=other)</sup>&nbsp;&nbsp;[![Chat with Dosu](https://dosu.dev/dosu-chat-badge.svg)](https://app.dosu.dev/32d7cdb1-783f-42b5-8007-7cac8945fe20/ask?utm_source=github)&nbsp;[![Join Discord](https://img.shields.io/badge/join-5865F2?logo=discord&logoColor=white&label=)](https://go.dosu.dev/discord-bot)&nbsp;[![Share on X](https://img.shields.io/badge/X-share-black)](https://twitter.com/intent/tweet?text=%40dosu_ai%20helped%20me%20solve%20this%20issue!&url=https%3A//github.com/opendatalab/MinerU/issues/3684)

**Comment by myhloli** [comment, marked as answer]
可以尝试将mineru和utils的版本都升级到最新，一次性推理1000页以上的文档并根据tqdm查看推理速度，单次推理页数过少无法通过batch获取速度提升

**Reply by taoweitao-125** [reply]
论文里的机器是单卡还是多卡呢，比如H200单卡下性能是4.47pages/s吗

**Comment by myhloli** [comment]
https://github.com/opendatalab/MinerU/discussions/3644

**Comment by taoweitao-125** [comment]
论文里的机器是单卡还是多卡呢，比如H200单卡下性能是4.47pages/s吗
